src/orchestrator-process/src/lib.rs

```diff
@@ -276,7 +277,8 @@ impl NamespacedOrchestrator for NamespacedProcessOrchestrator {
 
     async fn drop_service(&self, id: &str) -> Result<(), anyhow::Error> {
         let mut supervisors = self.supervisors.lock().expect("lock poisoned");
-        supervisors.remove(id);
+        // Explicitly drop the supervisor before returning from this function so that the temp files are removed.
+        drop(supervisors.remove(id));
```

Comment:
This looks equivalent to the existing code. Unless I'm missing something, would prefer to leave this as it was—the new comment seems to imply that there is something special happening here, but there isn't?